src/messages/HeadersMessage.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * This file is part of the bitcoin-classic project
+ * Copyright (c) 2009-2010 Satoshi Nakamoto
+ * Copyright (c) 2009-2015 The Bitcoin Core developers
+ * Copyright (C) 2017 Tom Zander <tomz@freedommail.ch>
+ * Copyright (C) 2017 Angel Leon <@gubatron>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "HeadersMessage.h"
+
+namespace {
+    bool HeadersMessage::handle(CNode *const pfrom,
+                                CDataStream &vRecv,
+                                int64_t nTimeReceived,
+                                std::string &strCommand,
+                                const bool xthinEnabled,
+                                const bool fReindexing) {
+        std::vector<CBlockHeader> headers;
+
+        // Bypass the normal CBlock deserialization, as we don't want to risk deserializing 2000 full blocks.
+        unsigned int nCount = ReadCompactSize(vRecv);
+        if (nCount > MAX_HEADERS_RESULTS) {
+            Misbehaving(pfrom->GetId(), 20);
+            return error("headers message size = %u", nCount);
+        }
+        headers.resize(nCount);
+        for (unsigned int n = 0; n < nCount; n++) {
+            vRecv >> headers[n];
+            ReadCompactSize(vRecv); // ignore tx count; assume it is 0.
+        }
+
+        LOCK(cs_main);
+
+        if (nCount == 0) {
+            // Nothing interesting. Stop asking this peers for more headers.
+            return true;
+        }
+
+        const CChainParams chainparams = Params();
+        CBlockIndex *pindexLast = NULL;
+        BOOST_FOREACH(const CBlockHeader &header, headers) {
+                        CValidationState state;
```

Comment:
Please be careful about your whitespace, we had this once before ;)

We use a really simple 4-spaces indent and never more than one indent for a block.

Reply:
will have to revise again my autoformatting, will fix.

Reply:
done.
Not sure why CLion might be doing this when I apply auto-format, everything has been set for 4 spaces, still looking into it.